analysis/study_definition.py

```diff
@@ -313,34 +456,97 @@
         include_date_of_match = True,
         date_format = "YYYY-MM-DD",
     ),
-    ## Reduced kidney function
-    creatinine = patients.with_these_clinical_events(
-        creatinine_codes, # imported from codelists.py
+    ### Dialysis
+    dialysis = patients.with_these_clinical_events(
+        dialysis_codes, # imported from codelists.py
+        returning = "binary_flag",
+        on_or_before = "index_date",
+        find_last_match_in_period = True,
+    ),
+    #### Date of dialysis
+    dialysis_date = patients.with_these_clinical_events(
```

Comment:
This is fine as you have it, but when you use `with_these_clinical_event`, you can set `include_date_of_match=False` in the `dialysis` var and this will return two variables - `dialysis` and `dialysis_date`. Saves having to copy the variable! Also works for `with_these_medications`

Reply:
This is great! I thought I needed to define dialysis_date as I'm using it directly to define another variable, but great I don't have to!